src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -845,7 +845,7 @@ public void WriteToJournal(LowLevelTransaction tx, int pageCount)
             if (CurrentFile == null || CurrentFile.AvailablePages < pages.NumberOfPages)
             {
                 _lazyTransactionBuffer?.WriteBufferToFile(CurrentFile, tx);
-                CurrentFile = NextFile(pages.NumberOfPages);
+                CurrentFile = NextFile((int)pages.NumberOfPages);
```

Comment:
There seems to be a repeating pattern of mixing the number of pages and the actual size.
int.MaxValue * 4KB = 16TB.